ros/src/twist_controller/dbw_node.py

```diff
@@ -81,20 +81,38 @@ def loop(self):
 
             if all([self.twist_cmd, self.current_velocity, self.dbw_enabled]):    # Ensure values have been initialized
 
+                ########## for testing purposes only, do not integrate into master ####################
+                #creates speed profile for testing longitudinal controller
+                decel_setpoint = 1
+                if rospy.get_time() - self.time_init < 30:
+                    speed_command = 13
+                elif rospy.get_time() - self.time_init < 60:
+                    speed_command = last_speed_command - .02*decel_setpoint    
+                    speed_command = max(speed_command, 0)   
+                elif rospy.get_time() - self.time_init < 90:
+                    speed_command = 8   
+                elif rospy.get_time() - self.time_init < 120:
+                    speed_command = last_speed_command - .02*decel_setpoint    
+                    speed_command = max(speed_command, 0) 
+                else: 
+                    speed_command = 11
+                last_speed_command = speed_command
+                ###################################################################################
+
                 # Get predicted throttle, brake and steering
-                throttle, brake, steering = self.controller.control(self.twist_cmd.linear.x,
-                    self.twist_cmd.angular.z, self.current_velocity.linear.x, self.dbw_enabled)
+                throttle, brake, steering = self.controller.control(speed_command,
+                    self.twist_cmd.angular.z, self.current_velocity.linear.x, self.dbw_enabled, self.log_handle) #self.twist_cmd.linear.x
 
                 # Log data for car control analysis
                 if self.log_to_csv:
-                    timestamp = rospy.get_rostime() - self.time_init
-                    self.log_data(timestamp.to_sec(), self.twist_cmd.linear.x, self.twist_cmd.angular.z,
-                                  self.current_velocity.linear.x, self.dbw_enabled, throttle, brake, steering)
+                    #timestamp = rospy.get_rostime() - self.time_init
```

Comment:
timestamp is used to get a smaller value, easier to understand in term of time since beginning of simulation (instead of very large number with no reference)

Reply:
I understand. Instead of doing this here, I do the same thing in the python_plotter script, which then allows to time align data from different ROS nodes.